packages/playground/website/src/lib/redux-store.ts

```diff
@@ -61,44 +69,52 @@ const initialState: AppState = {
 			? 'mount-markdown-directory'
 			: null,
 	offline: !navigator.onLine,
+	clients: {},
 	siteListing: {
 		status: { type: 'loading' },
 		sites: [],
 	},
 };
 
-if (query.get('storage') === 'browser') {
-	const siteSlug = query.get('site-slug') || 'wordpress';
-	const opfsRoot = await navigator.storage.getDirectory();
-	const opfsDir = await opfsRoot.getDirectoryHandle(
-		siteSlug === 'wordpress' ? siteSlug : 'site-' + siteSlug,
-		{
-			create: true,
-		}
-	);
-
-	initialState.opfsMountDescriptor = {
-		device: {
-			type: 'opfs',
-			path: await directoryHandleToOpfsPath(opfsDir),
-		},
-		mountpoint: '/wordpress',
-	};
-}
-
 // Create the slice
 const slice = createSlice({
 	name: 'app',
 	initialState,
-	selectors: {
-		getOpfsHandle: (state) => state.opfsMountDescriptor,
-	},
 	reducers: {
-		setPlaygroundClient: (
+		setActiveSite: (state, action: PayloadAction<SiteInfo>) => {
+			state.activeSite = action.payload;
+			state.opfsMountDescriptor = {
+				device: {
+					type: 'opfs',
+					path: '/' + getDirectoryNameForSlug(action.payload.slug),
+				},
+				mountpoint: '/wordpress',
+			};
+		},
+		forgetClientInfo: (state, action: PayloadAction<string>) => {
+			delete state.clients[action.payload];
+		},
+		setClientInfo: (
 			state,
-			action: PayloadAction<PlaygroundClient>
+			action: PayloadAction<{
+				siteSlug: string;
+				info: Partial<ClientInfo>;
+			}>
 		) => {
-			state.playgroundClient = action.payload;
+			const siteSlug = action.payload.siteSlug;
+			const clientInfo = action.payload.info;
+			if (!state.clients[siteSlug]) {
+				return;
```

Comment:
If we return early here when there isn't existing client info, how is client info set initially for a given slug?

Reply:
Ah, this is a bug and a leftover from my first pass on this function

Reply:
Or maybe it's not. I think I intended this to only do something when the `client` property is present so that the components may always assume every `clientInfo` comes with a `client`. There's likely a better way to enforce this constraint than a silent return.